packages/table/src/table.tsx

```diff
@@ -336,49 +336,13 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         selectionModes: SelectionModes.ALL,
     };
 
-    private static SHALLOWLY_COMPARABLE_PROP_KEYS = [
-        "enableFocus",
-        "allowMultipleSelection",
-        "children",
-        "fillBodyWithGhostCells",
-        "getCellClipboardData",
-        "isColumnReorderable",
-        "isColumnResizable",
-        "loadingOptions",
-        "onColumnWidthChanged",
-        "columnWidths",
-        "onColumnsReordered",
-        "isRowReorderable",
-        "isRowResizable",
-        "onRowHeightChanged",
-        "rowHeights",
-        "isRowHeaderShown",
-        "onRowsReordered",
-        "onSelection",
-        "onFocus",
-        "onCopy",
-        "renderRowHeader",
-        "renderBodyContextMenu",
-        "numRows",
-        "focusedCell",
-        // "selectedRegions" (intentionally omitted; can be deeply compared to save on re-renders in controlled mode)
-        "selectedRegionTransform",
-        "selectionModes",
-        "styledRegionGroups",
-    ];
-
-    private static SHALLOWLY_COMPARABLE_STATE_KEYS = [
-        "columnWidths",
-        "locator",
-        "isLayoutLocked",
-        "isReordering",
-        "viewportRect",
-        "verticalGuides",
-        "horizontalGuides",
-        "rowHeights",
-        // "selectedRegions" (intentionally omitted; can be deeply compared to save on re-renders in uncontrolled mode)
-        "focusedCell",
-    ];
+    private static SHALLOW_COMPARE_PROP_KEYS_BLACKLIST = [
+        "selectedRegions", // (intentionally omitted; can be deeply compared to save on re-renders in controlled mode)
+    ] as Array<keyof ITableProps>;
+
+    private static SHALLOW_COMPARE_STATE_KEYS_BLACKLIST = [
+        "selectedRegions", // (intentionally omitted; can be deeply compared to save on re-renders in uncontrolled mode)
+    ] as Array<keyof ITableState>;
```

Comment:
Seems like `Array<keyof X>` is required to let TS know that this isn't simply an array of `string`s. Here's the error that appears when you remove the explicit typing:

```
[ts]
Argument of type '{ exclude: string[]; }' is not assignable to parameter of type 'IKeyBlacklist<ITableProps & { children?: ReactNode; }> | IKeyWhitelist<ITableProps & { children?:...'.
  Type '{ exclude: string[]; }' is not assignable to type 'IKeyWhitelist<ITableProps & { children?: ReactNode; }>'.
    Property 'include' is missing in type '{ exclude: string[]; }'.
--------------------------
const propKeysBlacklist: {
    exclude: string[];
}
```

If we need to explicitly type, seems worthwhile to keep the two lists (unless we just define one list and cast it differently inline).